script/ExtractDeployedFoundryAddresses.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python3
+"""
+Script to extract deployed contract addresses from Foundry broadcast files.
+
+This script reads from the broadcast folder and generates a file with the latest deployed 
+smart contract addresses that are in the broadcast folder.
+
+It specifically looks at the run-latest.json file for each smart contract and inside 
+that JSON looks at the `contractAddress` field.
+"""
+
+import json
+import os
+import sys
+from datetime import datetime
+from pathlib import Path
+
+
+def find_broadcast_files(broadcast_dir: Path) -> list:
```

Comment:
Can this script run well in the github CI? I think that'd be an ideal end result of this script. Also I know you're probably more familiar with python but does this script require other dependencies? I am wary of adding more dependencies especially since we already have typescript set up..but its not a big deal. This file so far isn't part of the CI/hot path so its fine